clean.py

```diff
@@ -18,7 +18,9 @@
 MAPS = {
     "replace_newline_with_space": replace_newline_with_space,
     "remove_lines_with_code": build_bad_substring_remover(["{", "}", "[if", "<script"]), 
-    "remove_html_spans": build_bad_substring_remover(["<span", "</span>", "<div", "</div>", "<a", "</a>", "br>"])
+    "remove_html_spans": build_bad_substring_remover(["<span", "</span>", "<div", "</div>", "<a", "</a>", "br>"]),
+    "remove_html_spans_sanad": build_bad_substring_remover(["<img", "]]>", "<![CDATA", "//DW", "var ", "xtImg", "To view this video please enable JavaScript"]),
+    "remove_wiki_mojobake": build_bad_substring_remover(["À À À"])
```

Comment:
I'm surprised you use a set of 3.
Btw can we rename the `build_bad_substring_remove` to `build_bad_line_remover` given that we're now going to have a real substring remover.

Reply:
The google dedup filter had a set of 3 as a repeated substring.

```
Repeated in 9.46% of 82616 documents

À

Repeated in 3.28% of 82616 documents

À À

Repeated in 0.83% of 82616 documents

À À À
```

Reply:
Could you check a few samples where only one occurs? If there is garbage in 10% of the samples it is better to remove all of them.

Reply:
One is not enough:

'ನಿರೆನ್\u200dಬರ್ಗ್, ಮಾರ್ಷಲ್ 1927. ಅನುವಂಶಿಕತೆಯ ರಾಸಾಯನಿಕ ತಳಹದಿಯನ್ನು ಬಯಲು ಮಾಡಿದವರಲ್ಲಿ ಅಗ್ರಸ್ಥಾನವನ್ನು ಪಡೆದಿರುವ ಅಮೆರಿಕದ ಜೀವರಸಾಯನ ವಿಜ್ಞಾನಿ. 1948ರಲ್ಲಿ ಫ್ಲಾರಿಡ ವಿಶ್ವವಿದ್ಯಾಲಯದಿಂದ ಬಿ.ಎಸ್. ಡಿಗ್ರಿ ಪಡೆದ ನಿರೆನ್\u200dಬರ್ಗ್ ಮುಂದೆ ಅದೇ ವಿಶ್ವವಿದ್ಯಾಲಯದ ಎಂ.ಎಸ್.ಡಿಗ್ರಿ (ಜೀವಶಾಸ್ತ್ರ) ಕೂಡ ಪಡೆದರು. ಅನಂತರ ಮಿಷಿಗನ್ ವಿಶ್ವವಿದ್ಯಾಲಯದಲ್ಲಿ ಜೀವರಸಾಯನ ವಿಜ್ಞಾನದ ಕ್ಷೇತ್ರದಲ್ಲಿ ಸಂಶೋಧನೆ ನಡೆಸಿ ಪಿ.ಎಚ್.ಡಿ ಪದವಿಯನ್ನೂ ಗಳಿಸಿದರು. ಅವರ ಸಂಶೋಧನಾ ಚಟುವಟಿಕೆಗಳಿಗೆ ಅಮೆರಿಕನ್ ಕ್ಯಾನ್ಸರ್ ಸಂಸ್ಥೆ ಬೆಂಬಲವಿತ್ತು. ನ್ಯಾಷನಲ್ ಇನ್\u200dಸ್ಟಿಟ್ಯೂಟ್ ಆಫ್ ಹೆಲ್ತ್ ಸಂಸ್ಥೆಯಲ್ಲಿ ಕೆಲಸ ಮಾಡಲು ಪೋಸ್ಟ್ ಡಾಕ್ಟರೇಟ್ ಫೆಲೊಷಿಪ್ಪನ್ನು ನೀಡಿತು. ನಿರೆನ್\u200dಬರ್ಗ್ 1962ರಲ್ಲಿ ನ್ಯಾಷನಲ್ ಹಾರ್ಟ್ ಇನ್\u200dಸ್ಟಿಟ್ಯೂಟಿನ ಬಯೊಕೆಮಿಕಲ್ ಜೆನೆಜೆಕ್ಸ್ ಪ್ರಯೋಗಾಲಯದ ಮುಖ್ಯಾಧಿಕಾರಿಗಳಾದರು. ತಳಿಶಾಸ್ತ್ರವನ್ನು ಜೀವರಾಸಾಯನಿಕ ದೃಷ್ಟಿಯಿಂದ ಅಭ್ಯಸಿಸುವುದು ಈ ವಿಭಾಗದ ಮುಖ್ಯ ಉದ್ದೇಶ. ನಿರೆನ್\u200dಬರ್ಗ್ ಮತ್ತು ಅವರ ಸಹುದ್ಯೋಗಿಗಳು ನಡೆಸಿದ ಸಂಶೋಧನೆಗಳು ಈ ಕ್ಷೇತ್ರದಲ್ಲಿ ನೂತನ ಯುಗವನ್ನೇ ಅವಾಹಿಸಿದವು ಎಂದರೆ ತಪ್ಪಾಗಲಾರದು. ಈ ಸಂಶೋಧನೆಗಳ ಮಹತ್ತ್ವವನ್ನು ಗಮನಿಸಿ 1968ರಲ್ಲಿ ನಿರೆನ್\u200dಬರ್ಗ್ ಅವರಿಗೆ ನೊಬೆಲ್ ಪಾರಿತೋಷಿಕವನ್ನು ನೀಡಿ ಗೌರವಿಸಲಾಯಿತು. ಮೆಂಡಲ್ (1822-1884) ತಳಿಶಾಸ್ತ್ರಕ್ಕೆ ವಿಜ್ಞಾನದ ತಳಹದಿಯನ್ನು ಹಾಕಿದರೂ ಅನೇಕ ವರ್ಷಗಳ ಕಾಲ ಅದರ ರಾಸಾಯನಿಕ ಆಧಾರ ಸ್ಪಷ್ಟವಾಗಿರಲಿಲ್ಲ. ಈ ಶತಮಾನದ ಆದಿಯಲ್ಲಿ ಗೆರಾರ್ಡ್ ಎಂಬವರು ಅನುವಂಶಿಕವಾಗಿ ಕೊಡುಗೆಯಾಗುವ ಕೆಲವು ರೋಗಗಳಿಗೂ ಕೋಶಗಳಲ್ಲಿ ಉತ್ಪತ್ತಿಯಾಗುವ ಕಿಣ್ವಗಳಿಗೂ ಸಂಬಂಧವಿರಬೇಕೆಂದು ಗುರುತಿಸಿದರು. ವರ್ಣಹೀನತೆ, ಅಲ್\u200dಕ್ಯಾಪ್ಟೋನ್ ಯೂರಿಯ, ಸಿಕಲ್\u200dಸೆಲ್ ಅನೀಮಿಯ ಇತ್ಯಾದಿ ಅನುವಂಶಿಕ ರೋಗಗಳಿಗೆ, ಆ ರೋಗಿಗಳಲ್ಲಿ ವಿಶಿಷ್ಟವಾದ ಒಂದು ಕಿಣ್ವದ ಅಭಾವ ಕಾರಣ ಎಂದು ತಿಳಿದುಬಂದಿದೆ. ಕಿಣ್ವಗಳು ಪ್ರೋಟೀನುಗಳು. ಪ್ರೋಟೀನುಗಳು ಅಮೈನೋ ಆಮ್ಲಗಳೆಂಬ ಸರಳ ರಾಸಾಯನಿಕ ವಸ್ತುಗಳಿಂದ ರೂಪಿತವಾಗುವ ಬೃಹದಣುಗಳು. ಇವುಗಳ ಭೌತ, ರಾಸಾಯನಿಕ ಹಾಗೂ ಜೈವಿಕ ಗುಣಗಳೂ ಅಮೈನೊ ಆಮ್ಲಗಳ ಅನುಕ್ರಮವನ್ನು ಅವಲಂಬಿಸುತ್ತವೆ. ಪ್ರೋಟೀನುಗಳು ಜೀವಿಗಳ ಕೋಶದಲ್ಲಿ ಸಂಶ್ಲೇಷಣೆಯಾಗುತ್ತದೆ (ನೋಡಿ-ಪ್ರೋಟೀನುಗಳು). ಆದ್ದರಿಂದ ಪ್ರೋಟೀನುಗಳ ಸಂಶ್ಲೇಷಣೆಯ ಮೇಲೆ ಹತೋಟಿಯಿರುವ ಒಂದು ವ್ಯವಸ್ಥೆ ಕೋಶದಲ್ಲಿರಬೇಕು. ಕೋಶಗಳ ನ್ಯೂಕ್ಲಿಯಸ್ಸಿನಲ್ಲಿರುವ ವರ್ಣತಂತುಗಳು ರಾಸಾಯನಿಕವಾಗಿ ಪ್ರೋಟೀನನ್ನು ಒಳಗೊಂಡ ಡೀ ಆಕ್ಸಿ ರೈಬೊನ್ಯೂಕ್ಲಿಯಕ್ ಆಮ್ಲಗಳು (ಡಿ ಎನ್\u200dಎ). ಡಿಎನ್\u200dಎಗಳು ಅಡಿನೈನ್, ಗ್ವಾನೀನ್, ಸೈಟೋಸಿನ್, ಮತ್ತು ಥೈಮಿನ್ ಎಂಬ ಸಾರಜನಕಯುಕ್ತ ಪ್ರತ್ಯಾಮ್ಲಗಳು, ಡೀ ಆಕ್ಸಿ ರೈಬೋಸ್ ಎಂಬ ಸಕ್ಕರೆ, ಮತ್ತು ಫಾಸ್ಫೇಟ್ ಪುಂಜಗಳಿಂದ ರೂಪಿತವಾಗುವ ಬೃಹದಣುಗಳು. ಪ್ರೋಟೀನುಗಳಲ್ಲಿರುವ ವೈವಿಧ್ಯ ಡಿಎನ್\u200dಎ ಅಣುಗಳಲ್ಲಿಯೂ ಕಂಡುಬರುತ್ತದೆ. ಈ ಬಗೆಯ ವೈವಿಧ್ಯಕ್ಕೆ ಅವುಗಳಲ್ಲಿರುವ ಪ್ರತ್ಯಾಮ್ಲಗಳ ಮೊತ್ತ ಮತ್ತು ಅನುಕ್ರಮ ಕಾಋಣ. ಪ್ರೋಟೀನಿನ ಸಂಶ್ಲೇಷಣೆ ವರ್ಣವಸ್ತುಗಳ ಹತೋಟಿಯಲ್ಲಿ ಅಂದರೆ ಡಿಎನ್\u200dಎ ಅಣುಗಳ ಹತೋಟಿಯಲ್ಲಿ ಇರುವುದಾದರೆ ಪ್ರೋಟೀನುಗಳಲ್ಲಿರುವ ಅಮೈನೊ ಆಮ್ಲಗಳ ಅನುಕ್ರಮವನ್ನು ಡಿಎನ್\u200dಎ ನಿರ್ದೇಶಿಸುತ್ತದೆ ಎಂದಾಯಿತು. ಆದ್ದರಿಂದ ಡಿ ಎನ್ ಎ ಅಣುವಿನಲ್ಲಿರುವ ನೈಟ್ರೊಝನ್ ಪ್ರತ್ಯಾಮ್ಲಗಳ ಅನುಕ್ರಮಕ್ಕೂ ಪ್ರೋಟೀನುಗಳ ಅಮೈನೊ ಆಮ್ಲಗಳ ಅನುಕ್ರಮಕ್ಕೂ ಬಾಂಧವ್ಯವಿರಬೇಕೆಂಬುದು ಸ್ಪಷ್ಟ. ಡಿಎನ್\u200dಎ ಅಣುವಿನಲ್ಲಿ ಪ್ರತ್ಯಾಮ್ಲಗಳ ಸರಳ ನೀಳ ಅನುಕ್ರಮ, ಅಮೈನೊ ಆಮ್ಲಗಳ ಸಂಕೇತದಂತೆ ವರ್ತಿಸಬೇಕು. ಕ್ರಿಕ್ ಮತ್ತು ಬೆನ್ಸರ್ ಮತ್ತು ಇತರರ ಪ್ರಯೋಗಗಳಿಂದ ಮೂರು ಪ್ರತ್ಯಾಮ್ಲಗಳ ನೀಳ ಅನುಕ್ರಮ (ತ್ರಿವಳಿ) ಒಂದು ಅಮೈನೊ ಆಮ್ಲದ ಸಂಕೇತ ಎಂದು ತಿಳಿದು ಬಂದಿತು. ನಾಲ್ಕು ಪ್ರತ್ಯಾಮ್ಲಗಳನ್ನು ತ್ರಿವಳಿಗಳಂತೆ 64 ವಿಧಗಳಲ್ಲಿ ಜೋಡಿಸಬಹುದು. ಪ್ರೋಟೀನುಗಳಲ್ಲಿ ಸರಿಸುಮಾರಾಗಿ 21 ಬಗೆಯ ಅಮೈನೊ ಆಮ್ಲಗಳಿರುತ್ತವೆಯಾಗಿ ತ್ರಿವಳಿಗಳ ಸಂಖ್ಯೆ ಬೇಕಿರುವುದಕ್ಕಿಂತ ಹೆಚ್ಚಾಗಿಯೇ ಇದೆ. ದತ್ತ ತ್ರಿವಳಿ (ಉದಾಹರಣೆ ಂಉಉ) ಯಾವ ಅಮೈನೊ ಆಮ್ಲದ ಸಂಕೇತ ಎಂಬುದನ್ನು ತಿಳಿಯುವುದೇ ನಿರೆನ್\u200dಬರ್ಗ್ ಮತ್ತು ಅವರ ಸಹುದ್ಯೋಗಿಗಳು ಆಯ್ದುಕೊಂಡ ಕಾರ್ಯಕ್ಷೇತ್ರ. ನಿರೆನ್\u200dಬರ್ಗ್ ಮತ್ತು ಅವರ ಸಹುದ್ಯೋಗಿಗಳು ತಮ್ಮ ಸಂಶೋಧನೆಗಳನ್ನು 1960ರಿಂದ ಪ್ರಾರಂಭಿಸಿದರು. ಮೂರು ಉದ್ದೇಶಗಳು ಅವರ ಗಮನದಲ್ಲಿದ್ದವು; 1 ತ್ರಿವಳಿ ಸಂಕೇತಗಳಲ್ಲಿ ಅಡಿನೈನ್, ಗ್ವಾನೀನ್, ಸಐಟೋಸಿನ್ ಮತ್ತು ಥೈಮೀನುಗಳ ಪರಸ್ಪರ ಪ್ರಮಾಣ ಏನು? 2. ಯಾವ ತ್ರಿವಳಿ ಯಾವ ಅಮೀನೋ ಆಮ್ಲದ ಸಂಕೇತ, 3. ಯಾವುದೇ ನಿರ್ದಿಷ್ಟ ಅಮೈನೋ ಆಮ್ಲವನ್ನು ಸೂಚಿಸುವ ತ್ರಿವಳಿಯಲ್ಲಿ ಪ್ರತ್ಯಾಮ್ಲಗಳ ಅನುಕ್ರಮ ಏನು ? ಈ ಉದ್ದೇಶಗಳ ಸಾಧನೆಗೆ ಅವರು ಎಷರೀಕಿಯಕೋಲೈ ಎಂಬ ಸೂಕ್ಷ್ಮಜೀವಿಯಲ್ಲಿ ನಡೆಯುವ ಪ್ರೋಟೀನ್ ಸಂಶ್ಲೇಷಣೆಯನ್ನು ಅಭ್ಯಸಿಸಿದರು. ಇವುಗಳಲ್ಲಿ ಪ್ರೋಟೀನ್ ಸಂಶ್ಲೇಷಣೆ ಶೀಘÀ್ರವಾಗಿ ನಡೆಯುತ್ತದೆ ಮತ್ತು ಸಂಶ್ಲೇಷಣಾ ವ್ಯವಸ್ಥೆ ಪ್ರಾಣಿಕೋಶಗಳ ವ್ಯವಸ್ಥೆಗಿಂತ ಸರಳತರ. ಪ್ರಾಯೋಗಿಕ ದರಷ್ಟಿಯಿಂದ ಇಡೀ ಕೋಶವನ್ನು ಬಳಸುವುದಕ್ಕಿಂತ ಅದನ್ನು ಒಡೆದು ಘಟಕ ವಸ್ತುಗಳನ್ನು ಬೇರ್ಪಡಿಸಿ ಅವನ್ನು ಉಪಯೋಗಿಸುವುದು ಅನುಕೂಲ. ಇಂಥ ವ್ಯವಸ್ಥೆಯನ್ನು ಬಳಸಿ ನಡೆಸಿದ ಪ್ರಯೋಗಗಳಿಂದ ರೈಬೋಸೋಮುಗಳ ಮೇಲೆ ಪ್ರೋಟೀನಿನ ಸಂಶ್ಲೇಷಣೆ ನಡೆಯುವುದೆಂದೂ ಡಿ.ಎನ್.ಎ. ನೇತೃತ್ವದಲ್ಲಿ ಸಂಶ್ಲೇಷಿತವಾಗುವ ದೂತ ಆರ್.ಎನ್.ಎ. ಅಣು ಅಮೈನೋ ಆಮ್ಲಗಳು ತಮ್ಮ ತಮ್ಮ ಸ್ಥಾನವನ್ನು ಸೇರಲು ಮಾರ್ಗದರ್ಶನ ಮಾಡುವುದೆಂದೂ ತಿಳಿದುಬಂತು. ಅಮೈನೋ ಆಮ್ಲಗಳು ನೇರವಾಗಿ ತ್ರಿವಳಿ ಸಂಕೇತವನ್ನು ಗುರುತಿಸಲಾರವು. ಬದಲಾಗಿ ವಾಹಕ ಆರ್\u200dಎನ್\u200dಎ ಎಂಬ ರೈಬೋನ್ಯೂಕ್ಲಿಯಿಕ್ ಆಮ್ಲ ವಿಶೇಷಗಳು ಅಮೈನೋ ಆಮ್ಲಗಳನ್ನು ಹೊತ್ತು ರೈಬೋಸೋಮುಗಳ ಬಳಿ ಒಯ್ಯುತ್ತವೆ. ವಿಶಿಷ್ಟ ಅಮೈನೋ ಆಮ್ಲವೊಂದನ್ನು ವಿಶಿಷ್ಟ ವಾಹಕ ಆರ್\u200dಎನ್\u200dಎಯೊಂದು ಮಾತ್ರ ಒಯ್ಯುತ್ತದೆ. ದೂತ ಆರ್\u200dಎನ್\u200dಎ ಮಾತ್ರ ಹೊತ್ತಿರುವ ಅಮೈನೋ ಆಮ್ಲದ ಸಂಕೇತವಾದ ತ್ರಿವಳಿಯನ್ನು ಗುರುತಿಸಬಲ್ಲ ಪ್ರತಿ ತ್ರಿವಳಿ ವಾಹಕ ಆರ್\u200dಎನ್\u200dಎ ಅಣುವಿನ ಒಂದು ಭಾಗದಲ್ಲಿ ಇರುತ್ತದೆ. ದೂತ ಆರ್\u200dಎನ್\u200dಎ ಅಣುವಿನಲ್ಲಿ ತ್ರಿವಳಿಗಳು ಒಂದಾದ ಮೇಲೆ ಒಂದರಂತೆ ಇರುವುದರಿಂದ ತತ್ಸಂಬಧಿತ ಅಮೈನೋ ಆಮ್ಲಗಳು ಒಂದಾದ ಮೇಲೆ ಒಂದರಂತೆ ಪೆಪ್ಪಟೈಡ್ ಬಂಧಗಳಿಂದ ಬದ್ಧವಾಗಿ ಪ್ರೋಟೀನ್ ಅಣು ಸೃಷ್ಟಿಯಾಗುತ್ತದೆ. ದೂತ ಆರ್\u200dಎನ್\u200dಎ, ನ್ಯೂಕ್ಲಿಯಸ್ಸಿನಲ್ಲಿರುವ ಡಿಎನ್.ಎ. ಅಣುವಿನಲ್ಲಿರುವ ಪ್ರತ್ಯಾಮ್ಲಗಳ ಅನುಸಾರವಾಗಿ ಯಾವುದೇ ಬಗೆಯ ತಪ್ಪುಗಳಾಗದಂತೆ ಸಂಶ್ಲೇಷಿತವಾಗುತ್ತದೆ. ಈ ಸಂಶ್ಲೇಷಣೆ ಪ್ರತ್ಯಾಮ್ಲ ಜೋಡಿಗಳ ನಿಯಮಕ್ಕೆ ಅನುಸಾರವಾಗಿ ನಡೆಯುತ್ತದೆ. ಅಂದರೆ, ಡಿಎನ್\u200dಎನಲ್ಲಿ ಅಡಿನೈನ್ ಇದ್ದರೆ, ದೂತ ಆರ್\u200dಎನ್\u200dಎನಲ್ಲಿ ಅದಕ್ಕೆ ಹೊಂದಿಕೊಳ್ಳುವ ಸ್ಥಾನದಲ್ಲಿ ಯುರಾಸಿಲ್,. ಗ್ವಾನಿನ್ ಇದ್ದರೆ ಸೈಟೋಸಿನ್, ಸೈಟೋಸಿನ್ ಇದ್ದರೆ ಗ್ವಾನಿನ್, ಥೈಮಿನ್ ಇದ್ದರೆ ಅಡಿನೈನ ಸೇರುತ್ತವೆ: G-A-C-T-G-C-A-A-A...... ಊಹಾತ್ಮಕ ಡಿ ಎನ್ ಎ ಸರಣಿ C-T-G-A-C-G-T-T-T ........ (ಸಂಶ್ಲೇಷಿತವಾಗುವ ದೂತ ಆರ್\u200dಎನ್\u200dಎ) (ಆರ್ ಎನ್ ಎ ಅಣುವಿನಲ್ಲಿ ಥೈಮಿನ್ ಇರುವುದಿಲ್ಲ. ಬದಲಾಗಿ ಯುರಾಸಿಲ್ ಇರುತ್ತದೆ; ಡೀಆಕ್ಸಿರೈಬೋಸ್ ಬದಲು ರೈಬೋಸ್ ಇರುತ್ತದೆ). ವಾಹಕ ಆರ್\u200dಎನ್\u200dಎಗಳು ದೂತ ಆರ್\u200dಎನ್\u200dಎ ಅಣುವಿನ ತ್ರಿವಳಿಗಳನ್ನು ಇದೇ ತತ್ತ್ವದ ಆಧಾರದಿಂದ ಗುರುತಿಸುತ್ತದೆ. ಉದಾಹರಣೆಗೆ ಅ-U-ಉ ತ್ರಿವಳಿಯ ಪ್ರತಿ ಸಂಕೇತ ಉ-ಂ-ಅ ಇದು ಡಿಎನ್\u200dಎ ಅಣುವಿನ ಸಂಕೇತ ಉಂಅ ಗೆ ಸರಿಹೊಂದುತ್ತದೆ. ಪ್ರೋಟೀನ್ ಸಂಶ್ಲೇಷಣೆಯಲ್ಲಿ ರೈಬೋಸೋಮುಗಳನ್ನು ದೂತ ಆರ್\u200dಎನ್\u200dಎ ಹಗ್ಗದಂತೆ ಬಂಧಿಸುತ್ತದೆ. ಅಮೈನೋ ಆಮ್ಲಗಳನ್ನು ಹೊತ್ತ ವಾಹಕ ಆರ್\u200dಎನ್\u200dಎ ತನ್ನ ಪ್ರತಿ ತಿವಳಿಗೆ ಸರಿಹೊಂದುವ ಸ್ಥಾನವನ್ನು ದೂತ ಆರ್\u200dಎನ್\u200dಎಯಲ್ಲಿರುವ ತ್ರಿವಳಿಗೆ ಅನುಸಾರವಾಗಿ ಪತ್ತೆ ಹಚ್ಚಿ ತಾನೂ ರೂನೋಸೋಮುಗಳಿಗೆ ಬದ್ಧವಾಗುತ್ತದೆ. ಈ ತತ್ತ್ವದ ಆಧಾರದಿಂದ ಸಂಕೇತದ ರಹಸ್ಯವನ್ನು ಬಯಲು ಮಾಡಲು ಸಾಧ್ಯವಾಯಿತು. ಪ್ರಾರಂಭದಲ್ಲಿ ನಿರೇನ್\u200dಬರ್ಗ್ ಎಷರೀಕಿಯಾ ಕೋಲೈ ಕೋಶರಹಿತ ವ್ಯವಸ್ಥೆಗೆ ಬೇರೆ ಮೂಗಳಿಂದ ಪಡೆದ ಆರ್\u200dಎನ್\u200dಎಗಳು ಮತ್ತು ವಿಕಿರಣಪಟು ಇಂಗಾಲವನ್ನು ಒಳಗೊಂಡ ಅಮೈನೋ ಆಮ್ಲ ಒಂದು ಸಾಧಾರಣ ಅಮೈನೋ ಆಮ್ಲಗಳು ಇವನ್ನು ಸೇರಿಸಿ ಪ್ರೋಟೀನ್ ಸಂಶ್ಲೇಷಣೆಯನ್ನು ಪ್ರೇರೇಪಿಸಿದರು. ಅನಂತರ ಇದೇ ವ್ಯವಸ್ಥೆಯನ್ನು ಸರಳಗೊಳಿಸಲು ಕೃತಕವಾಗಿ ತಯಾರಿಸಿದ ದೂತ ಆರ್\u200dಎನ್\u200dಎಗಳನ್ನು ಬಳಸಿ ಯಾವೊಂದು ಉದ್ದೇಶಿತ ಅಮೈನೋ ಆಮ್ಲವನ್ನು ಪೆಪ್ಟೈಡ್ ಬಂಧದಲ್ಲಿ ತೊಡಗುವಂತೆ ಮಾಡಲು ಸಾಧ್ಯವೇ ಎಂದು ಪರೀಕ್ಷಿಸಲಾಯಿತು. ಕೃತಕ ದೂತ ಆರ್\u200dಎನ್\u200dಎಯಲ್ಲಿ ಪ್ರತ್ಯಾಮ್ಲಗಳ ಪ್ರಮಾಣವನ್ನು ಇಷ್ಟಾನುಸಾರವಾಗಿ ಬದಲಾಯಿಸುವ ಸಾಧ್ಯತೆ ಇರುವುದರಿಂದ ಮತ್ತು ಅವುಗಳ ಪ್ರಮಾಣವನ್ನು ನಿರ್ಧರಿಸುವುದು ಸುಲಭವಾದ್ದರಿಂದ ಈ ಬಗೆಯ ತಂತ್ರ ಪ್ರಯೋಗ ದೃಷ್ಟಿಯಿಂದ ಉತ್ತಮವಾದದ್ದು. ಹೀಗೆ ನಡೆಸಿದ ಪ್ರಯೋಗವನ್ನು ಕೇವಲ ಯುರಾಸಿಲ್\u200dನ್ನು ಮಾತ್ರ ಒಳಗೊಂಡ ಕೃತಕ ಆರ್\u200dಎನ್\u200dಎಯನ್ನು ಬಳಸಿದಾಗ 21 ಆಮೈನೋ ಆಮ್ಲಗಳ ಪೈಕಿ ಫಿನೈಲ್ ಅನಾನಿಲ್ ಮಾತ್ರ ಪೆಪ್ಟೇಡ್ ಬಂಧಗಳಲ್ಲಿ ತೊಡಗುತ್ತದೆ ಎಂದು ಪತ್ತೆಯಾಯಿತು. ಎಂದರೆ- UUU - ತ್ರಿವಳಿ, ಫೀನೈಲ್ ಅಲಾನಿನ್ನಿನ ಸಂಕೇತವಿರಬೇಕೆಂದು ಊಹಿಸಬಹುದು. ಇದೇ ರೀತಿ ಸೈಟೋಸಿನ್ ಮಾತ್ರ ಇರುವ ಬಹ್ವಂಶಿಯನ್ನು ಬಳಸಿದಾಗ ಪ್ರೋಲೀನ್ ಮಾತ್ರ ಇರುವ ಪ್ರೋಟೀನ್ ದೊರೆಯಿತು. ಇದರಿಂದ ಅಅಅ ಎಂಬುದು ಪ್ರೋಲೋನಿನ ಸಂಕೇತವಿರಬೇಕೆಂದಾಯಿತು. ಈ ಫಲಿತಾಂಶÀಗಳಿಂದ ಉತ್ತೇಜಿತಗೊಂಡ ನಿರೆನ್\u200dಬರ್ಗ್ ಪ್ರತ್ಯಾಮ್ಲಗಳಾದ ಂ, ಉ, ಅ, ಮತ್ತು Uಗಳನ್ನು ಸಾಧ್ಯವಾದ ಎಲ್ಲರೀತಿಗಳಲ್ಲಿ ವಿವಿಧ ಪ್ರಮಾಣಗಳಲ್ಲಿ ಜೋಡಿಸಿ ಹಲವಾರು ಕೃತಕ ದೂತ ಆರ್\u200dಎನ್\u200dಎಗಳನ್ನು ಸೃಷ್ಟಿಸಿದರು. ಕೋಶರಹಿತ ವ್ಯವಸ್ಥೆಗಳಲ್ಲಿ ಇವನ್ನು ದೂತ ಆರ್\u200dಎನ್\u200dಎಗಳಂತೆ ಬಳಸಿ ದೊರೆತ ಫಲಿತಾಂಶಗಳನ್ನು ಸಂಖ್ಯಾಕಲನಾತ್ಮವಾಗಿ ಪರಿಶೀಲಿಸಿ ಸಂಕೇತಗಳ ನಿಘಂಟನ್ನೇ ತಯಾರಿಸಿದರು. ಸಂಕೇತಗಳು ನಿರ್ದಿಷ್ಟವಾಗಿರದೇ ಸ್ಖಾಲಿತ್ಯವನ್ನು ಪ್ರದರ್ಶಿಸುತ್ತವೆ. ಎಂದರೆ ಒಂದೇ ಅಮೈನೋ ಆಮ್ಲವನ್ನು ಒಂದಕ್ಕಿಂತ ಹೆಚ್ಚಾದ ತ್ರಿವಳಿಗಳು ಸೂಚಿಸುತ್ತವೆ. ಉದಾಹರಣೆಗೆ ಪ್ರೋಲೀನ್ ಅಮೈನೋ ಆಮ್ಲವನ್ನು ಅಅU, ಅಅಅ, ಅಅಂ ಅಥವಾ ಅಅಉಗಳ ಪೈಕಿ ಯಾವುದೇ ಒಂದು ಜೋಡಣೆ ಸೂಚಿಸುತ್ತದೆ. ಹಲವು ಜೋಡಣೆಗಳು ಯಾವುದೇ ಅಮೈನೋ ಆಮ್ಲವನ್ನು ಸೂಚಿಸುವುದಿಲ್ಲ. ಅವು ಅರ್ಥರಹಿತ ಸಂಕೇತಗಳು. ದೂತ ಆರ್\u200dಎನ್\u200dಎ ಸಂಕೇತಗಳು ಅಮೈನೋ ಆಮ್ಲದ ಭಾಷೆಗೆ ತರ್ಜುಮೆಯಾಗುವ ಇರುವ ಮಿರಾಮ ಚಿಹ್ನೆಗಳೇ ಇಂಥ ಅರ್ಥರಹಿತ ಸಂಕೇತಗಳು ಇರಬಹುದೆಂದು ಊಹಿಸಲಾಯಿತು. ಎಂದರೆ, ಈ ಪ್ರದೇಶಗಳು ಪ್ರೋಲೀನಿನ ಸಂಶ್ಲೇಷಣೆ ಇಲ್ಲಿಂದ ಪ್ರಾರಂಭವಾಗಲಿ, ಇಲ್ಲಿ ನಿಲ್ಲಲಿ ಎಂಬ ಸೂಚನೆಗಳನ್ನೀಯಬಹುದು. ನಿರೇನ್\u200dಬರ್ಗ್ ತಮ್ಮ ಪ್ರಯೋಗಗಳನ್ನು ಎಷ್ಟು ಎಚ್ಚರಿಕೆಯಿಂದ ನಡೆಸಿ ದೊರೆತ ಫಲಿತಾಂಶಗಳನ್ನು ಎಷ್ಟು ಸರಿಯಾಗಿ ತರ್ಕಿಸಿದರೆಂದರೆ, ಅವರು ನಿರ್ಧರಿಸಿದ 50 ತ್ರಿವಳಿ ಸಂಕೇತಗಳ ಪೈಕಿ 45 ಸಂಕೇತಗಳು ಸತ್ಯ ಎಂದು ಅನಂತರ ನಡೆಸಿದ ಪ್ರಯೋಗಗಳು ಖಚಿತಗೊಳಿಸಿದವು. ಮೇಲಿನ ಪ್ರಯೋಗಗಳು ಯಾವೊಂದು ಅಮೈನೋ ಆಮ್ಲವನ್ನು ನಿರ್ದೇಶಿಸುವ ಸಂಕೇತದಲ್ಲಿರುವ ಪ್ರತ್ಯಾಮ್ಲಗಳ ಪ್ರಕೃತಿ ಮತ್ತು ಪರಸ್ಪರ ಪ್ರಮಾಣಗಳನ್ನು ಮಾತ್ರ ತಿಳಿಯಲು ಸಾಧ್ಯವಾಯಿತೇ ವಿನಾ ತ್ರಿವಳಿಯಲ್ಲಿರುವ ಪ್ರತ್ಯಾಮ್ಲಗಳು ಯಾವ ಅನುಕ್ರಮದಲ್ಲಿವೆ ಎಂಬುದನ್ನು ತಿಳಿಯಲಾಗಲಿಲ್ಲ. ಮುಂದಿನ ಹಂತದಲ್ಲಿ ನಿರೇನ್\u200dಬರ್ಗ್ ಮತ್ತು ಸಹೋದ್ಯೋಗಿಗಳು ಗೊತ್ತಾದ ಅನುಕ್ರಮದಲ್ಲಿರುವ ಕೃತಕ ತ್ರಿವಳಿಗಳನ್ನು ಸಂಶ್ಲೇಷಿಸಿ ಕೋಶರಹಿತ ವ್ಯವಸ್ಥೆಯಲ್ಲಿ ಇವನ್ನು ದೂತ ಆರ್\u200dಎನ್\u200dಎ ಬದಲಾಗಿ ತಿಳಿಸಿದರು. ಪ್ರಯೋಗ ವ್ಯವಸ್ಥೆಯಲ್ಲಿ ಎಲ್ಲಾ ಅಮೈನೋ ಅಮ್ಲಗಳು, ಇವುಗಳಿಗೆ ವಿಶಿಷ್ಟವಾದ ವಾಹಕ ಆರ್ ಎನ್ ಎ ಹಾಗೂ ರೈಬೋಸೋಮುಗಳು ಮತ್ತು ಇತರ ಸಹಾಯಕ ವಸ್ತುಗಳನ್ನು ಅಳವಡಿಸಲಾಗಿತ್ತು. ಅಮೈನೋ ಆಮ್ಲಗಳ ಪೈಕಿ ಒಂದು ಅಮೈನೋ ಆಮ್ಲದಲ್ಲಿ ವಿಕಿರಣಪಟು ಕಾರ್ಬನ್ (14ಚ) ಇರುವಂತೆ ವ್ಯವಸ್ಥೆ ಮಾಡಲಾಗಿತ್ತು. ಇಂತಹ ವ್ಯವಸ್ಥೆಗೆ ಪೂರ್ವಯೋಜಿತ ಅನುಕ್ರಮವುಳ್ಳ ತ್ರಿವಳಿಯನ್ನು ಸೇರಿಸಿದಾಗ ತ್ರಿವಳಿಯಿಂದ ನಿದೇರ್ಶಿತವಾದ ಅಮೈನೋ ಆಮ್ಲವನ್ನು ಹೊತ್ತ ವಾಹಕ ಆರ್. ಎನ್. ಎ. ಮಾತ್ರ ರೈಬೋಸೋಮುಗಳಿಗೆ ಬದ್ಧವಾಗುತ್ತದೆ ಎಂದು ನಿರೀಕ್ಷಿಸಬಹುದು. ಯಾವ ಅಮೈನೋ ಆಮ್ಲ ಈ ರೀತಿ ಬದ್ಧವಾಗುತ್ತದೆ ಎಂಬುದನ್ನು ನಿರೇನ್\u200dಬರ್ಗ್ ವಿಶಿಷ್ಟ ತಂತ್ರವೊಂದರಿಂದ ಪತ್ತೆ ಹಚ್ಚಿದರು. ಇದರಿಂದ ಹೆಚ್ಚು ಕಡಿಮೆ ಎಲ್ಲ 64 ತ್ರಿವಳಿಗಳ ಸಂದರ್ಭದಲ್ಲಿಯೂ ಸಂಕೇತದ ರಹಸ್ಯವನ್ನು ಬಯಲು ಮಾಡಲು ಸಹಾಯವಾಯಿತು. ಈ ಫಲಿತಾಂಶಗಳ ಬೆಳಕಿನಲ್ಲಿ ನಿರೆನ್\u200dಬರ್ಗ್ ತ್ರಿವಳಿಗಳನ್ನು ಗುರುತಿಸುವ ಪ್ರಕ್ರಮ, ವೈರಸ್ಸುಗಳು ಬ್ಯಾಕ್ಟೀರಿಯಾಗಳ ಮೇಲೆ ದಾಳಿ ನಡೆಸಿದಾಗ ನಡೆತಯುವ ಪ್ರೋಟೀನ್ ಸಂಶ್ಲೇಷಣೆಯ ಹತೋಟಿ ಕ್ರಮ, ಭ್ರೂಣದಲ್ಲಿ ಕೋಶಗಳು ವಿಶಿಷ್ಟ ವಿಭಜನೆಗಳಾಗುವ ರೀತಿ ಇತ್ಯಾದಿ ಕ್ಷೇತ್ರಗಳಲ್ಲಿ ಗಮನಾರ್ಹವಾದ ಪ್ರಗತಿಯನ್ನು ಸಾಧಿಸಿದರು. (ಎಚ್.ಎಸ್.ಎಸ್)'

>

Reply:
But two seems to be. 

133 documents have "À À" in them, and 118 have "À À À"